server/backend/testspg/crud/test_crud_permission_audit.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import DataError
+from api.app.crud.crud_permission_audit import (
+    read_permission_audit_history_by_user_and_application,
+)
+from testspg.fixture.permission_audit_fixture import (
+    APPLICATION_ID_1,
+    APPLICATION_ID_2,
+    AUDIT_RECORD_U1_A1_D1,
+    AUDIT_RECORD_U1_A1_D2,
+    AUDIT_RECORD_U1_A2,
+    AUDIT_RECORD_U2_A2,
+    USER_ID_1,
+    PERFORMER_DETAILS_1,
+)
+
+
+# No Records
+def test_read_permission_audit_history_no_records(db_pg_session: Session):
+    user_id = 999
+    application_id = 999
+
+    result = read_permission_audit_history_by_user_and_application(
+        user_id, application_id, db_pg_session
+    )
+
+    assert len(result) == 0
+
+
+# No Matching Records
+def test_read_permission_audit_history_no_matching_records(db_pg_session: Session):
+    user_id = 999
+    application_id = 999
+
+    result = read_permission_audit_history_by_user_and_application(
+        user_id, application_id, db_pg_session
+    )
+
+    assert result == []
+
+
+# Invalid Data Types
+def test_read_permission_audit_history_invalid_data_types(db_pg_session: Session):
+    with pytest.raises(DataError):
+        read_permission_audit_history_by_user_and_application(
+            "invalid_user_id", APPLICATION_ID_1, db_pg_session
+        )
+
+
+# Multiple Users, Same Application
+def test_read_permission_audit_history_multiple_users_same_application(
+    db_pg_session: Session,
+):
+    db_pg_session.add(AUDIT_RECORD_U1_A2)
+    db_pg_session.add(AUDIT_RECORD_U2_A2)
+    db_pg_session.commit()
```

Comment:
Hi Craig, wondering why `.commit()` is needed? I guess if you commit(), the test container won't be able to roll back, then in some other tests scenarios later it could have strange result.

Reply:
https://github.com/bcgov/nr-forests-access-management/pull/1587/commits/2a265190d616d1ba1630fd41c808eb19268ef2a9

Honestly i don't think it matters, because after adding the id to the record fixtures, the tests that insert the same record works fine with `.commit()`, but i removed it and they still work

Reply:
Thanks for trying!!